### DISCUSSION TITLE
How long does the master token last?

### DISCUSSION THREAD

**Discussion by thejoeosborne** [mxrch/GHunt#491] [Q&A] [answered]
Curious how long the master token lasts. @mxrch mentioned here that it "(almost) never expires" but I'm wondering if anyone knows the exact time frame that it expires.

**Comment by mxrch** [comment, marked as answer]
Hey !
From the article where I learnt about it : https://ethanblake4.medium.com/how-i-stole-the-data-in-millions-of-peoples-google-accounts-aa1b72dcc075

> The master token never expires, unless the user changes their password or two-factor settings. It’s not subject to any security checks based on location, IP, or activity, as best as I can tell. It never results an email or notification being sent to the user.

**Reply by thejoeosborne** [reply]
Thank you!!
